cdc/sinkv2/eventsink/txn/txn_sink.go

```diff
@@ -14,24 +14,197 @@
 package txn
 
 import (
+	"encoding/binary"
+	"hash/crc64"
+	"sync"
+
+	"github.com/pingcap/log"
 	"github.com/pingcap/tiflow/cdc/model"
 	"github.com/pingcap/tiflow/cdc/sinkv2/eventsink"
+	"github.com/pingcap/tiflow/pkg/causality"
+	"go.uber.org/zap"
+)
+
+const (
+	conflictDetectorSlots int64 = 1024 * 1024
 )
 
 // Assert EventSink[E event.TableEvent] implementation
 var _ eventsink.EventSink[*model.SingleTableTxn] = (*Sink)(nil)
 
 // Sink is the sink for SingleTableTxn.
-type Sink struct{}
+type Sink struct {
+	conflictDetector *causality.ConflictDetector[*worker, txnEvent]
+	workers          []*worker
+}
+
+func newSink(bes []backend) Sink {
+	var workers []*worker = make([]*worker, 0, len(bes))
+	for i, be := range bes {
+		w := newWorker(i, be)
+		w.wg.Add(1)
+		go w.Run()
+		workers = append(workers, w)
+	}
+	detector := causality.NewConflictDetector[*worker, txnEvent](workers, conflictDetectorSlots)
+	return Sink{detector, workers}
+}
 
 // WriteEvents writes events to the sink.
 func (s *Sink) WriteEvents(rows ...*eventsink.TxnCallbackableEvent) {
-	// TODO implement me
-	panic("implement me")
+	for _, row := range rows {
+		_ = s.conflictDetector.Add(txnEvent{row})
+	}
 }
 
-// Close closes the sink.
+// Close closes the sink. It won't wait for all pending items be handled.
 func (s *Sink) Close() error {
-	// TODO implement me
-	panic("implement me")
+	for _, w := range s.workers {
+		w.Close()
+	}
+	s.conflictDetector.Close()
+	return nil
+}
+
+type txnEvent struct {
+	*eventsink.TxnCallbackableEvent
+}
+
+// ConflictKeys implements causality.txnEvent interface.
+func (e txnEvent) ConflictKeys() (sums []int64) {
+	keys := genTxnKeys(e.TxnCallbackableEvent.Event)
+	sums = make([]int64, 0, len(keys))
+	for _, key := range keys {
+		hasher := crc64.New(crc64.MakeTable(crc64.ISO))
```

Comment:
Is it necessary to create a new hasher each time? A benchmark case is probably helpful in determining whether the performance is satisfactory.